gelf/tcpwriter.go

```diff
@@ -87,3 +63,19 @@ func (w *TCPWriter) Write(p []byte) (n int, err error) {
 
 	return len(p), nil
 }
+
+func (w *TCPWriter) writeToSocketWithReconnectAttempts(zBytes []byte) (n int, err error) {
+	var errConn error
+	w.mu.Lock()
+	for i := 0; n <= 3; i++ {
```

Comment:
I fixed the logic and made the maximum number of reconnection attempts configurable. I am uncertain how the logging should work however. Simply returning the number of bytes **and** an error would probably be considered by the client as a failure. Changing the method signature would mean not implementing the io.Writer interface. Do you have an idea I might have missed?